fleetspeak/src/client/https/polling_test.go

```diff
@@ -372,6 +377,87 @@ G:
 	cl.Stop()
 }
 
+func TestCommunicator(t *testing.T) {
+	testCommunicator(t, nil)
+}
+
+// A simple HTTP handler implementing a HTTPS proxy.
+// It implements the connect method only.
+type proxyHandler struct {
+	t *testing.T
+	// Number of requests handled.
+	numRequests uint32
+}
+
+// Implements a HTTPS proxy.
+// Accepts only the HTTP connect method.
+// Handles the request by hijacking the HTTP connection, opening a second connection to the host of
+// the request and copying data between the two connections.
+func (ph *proxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	t := ph.t
+	d, err := httputil.DumpRequest(r, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("Got proxy request: %s.", string(d))
+	if r.Method != http.MethodConnect {
+		t.Fatalf("Proxy received invalid method: %s.", r.Method)
+	}
+	conn, err := net.Dial("tcp", r.Host)
+	if err != nil {
+		t.Fatal(err)
+	}
+	w.WriteHeader(http.StatusOK)
+	hijacker, ok := w.(http.Hijacker)
+	if !ok {
+		t.Fatal("Failed to hijack HTTP connection.")
+	}
+	httpConn, _, err := hijacker.Hijack()
+	if err != nil {
+		t.Fatal(err)
+	}
+	atomic.AddUint32(&ph.numRequests, 1)
+	c := make(chan bool)
```

Comment:
nit

A channel for empty structs is a more popular choice for what you want to do here. It would convey more explicitly the fact that the value passed through the channel isn't important in and of itself (e.g you wouldn't need to pass 'false' via the channel).

https://dave.cheney.net/2014/03/25/the-empty-struct

Reply:
Done.